### PR TITLE
Improve generateDependencyList performance

### PR DESCRIPTION
**Pull request by riyafa**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by warunalakshitha** [comment]
Do we have any perf numbers with the fix?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmPackageGen.java [hunk 1]
```diff
@@ -131,14 +130,14 @@
...
-    private final Map<String, PackageID> dependentModules;
+    private final Set<PackageID> dependentModules;
     private final BLangDiagnosticLog dlog;
 
     JvmPackageGen(SymbolTable symbolTable, PackageCache packageCache, BLangDiagnosticLog dlog) {
         birFunctionMap = new HashMap<>();
         globalVarClassMap = new HashMap<>();
         externClassMap = new HashMap<>();
-        dependentModules = new LinkedHashMap<>();
+        dependentModules = new LinkedHashSet<>();
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmPackageGen.java:140]: Shall we add the perf numbers before and after the fix?

Comment (riyafa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmPackageGen.java:140]: I did the change based on the flame graph. The `getPackageName` consumed most of the CPU in the  `generateDependencyList` method. By this change we remove this call, the cpu consumption of `generateDependencyList` was dropped to that was before the encoding feature was introduced.

**Comment by manuranga** [comment]
Pr need to be rebased.

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**